### DISCUSSION TITLE
When we upgrade modules should be always ensure all namespaces have to to be tagged with latest versions?

### DISCUSSION THREAD

**Discussion by AntonyBishop** [ministryofjustice/cloud-platform#2728] [General]
Question: **When we upgrade modules should be always ensure all namespaces have to to be tagged with latest versions?**

We be good to understand what we do now, pros, cons, and alternatives?

Discuss!

**Comment by davidread** [comment]
I'd appreciate it someone can explain a bit about this question, for my benefit

**Reply by razvan-moj** [reply]
my take on this: if we keep the current rule, which is "en every release, the tag is changed for all using instances in the environments repo" this can be 

- a bit nerve-wrecking
- discouraging breaking but useful changes

**Reply by jasonBirchall** [reply]
@davidread Currently, every release to a Terraform module (i.e. RDS) results in us having to make a commit in cloud platform namespaces that contains an RDS call to the module. Sometimes, as with RDS, you can be touching/upgrading as many as 150 RDS instances at once. Which is rather scary, as Raz mentions.

**Comment by razvan-moj** [comment]
Con: pinned modules can stray too much from current releases, blocking upgrades of the terraform binary or providers.

**Reply by razvan-moj** [reply]
a possible fix is to have all the needed binaries locally, feature possible since terraform 0.14

**Reply by jasonBirchall** [reply]
We manage out of date modules via the [reports](https://reports.cloud-platform.service.justice.gov.uk/terraform_modules) page. I feel this is quite effective and essentially nags you to upgrade.

**Reply by razvan-moj** [reply]
I feel maybe it nags the wrong channel, and teams may be more interested to fix/upgrade themselves if they have visibility into the reasons why the new release exists.

**Comment by jasonBirchall** [comment]
What would be the alternative to changing the pin in a module call? latest/main?

This brings its own set of problems.
